lib/publish.js

```diff
@@ -102,8 +102,9 @@ export default async (pluginConfig, context) => {
               throw error;
             }
 
-            const { url } = response.file;
-
+            //const { url } = response.file;
+	    const url = uploadEndpoint;   
```

Comment:
The `uploadEndpoint` also includes a [`status` and `select` query](https://github.com/semantic-release/gitlab/blob/master/lib/publish.js#L91-L95) which is not required for the download. I would suggest to create a `genericsEndpoint` which is basically the `downloadEndpoint` and enrich it with the query for the upload. Something like:
```js
const genericsEndpoint= urlJoin(
  gitlabApiUrl,
  `/projects/${encodedRepoId}/packages/generic/release/${encodedVersion}/${encodedLabel}`
);

const uploadEndpoint = `${genericsEndpoint}?${status ? `status=${status}&` : ""}select=package_file`;
```